components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -101,9 +104,15 @@ export const WorkspaceEntry: FunctionComponent<Props> = ({ info, shortVersion })
 };
 
 export function getProjectPath(ws: Workspace) {
-    if (CommitContext.is(ws.context)) {
-        return `${ws.context.repository.host}/${ws.context.repository.owner}/${ws.context.repository.name}`;
-    } else {
-        return undefined;
+    if (!ws.status?.gitStatus?.cloneUrl) {
+        return ws.contextUrl.replace("https://", "");
     }
+    // http[s]?:\/\/(?<project>.*?\/.*?\/.*?)[./]
+    const regex = /http[s]?:\/\/(?<host>.*?)\/(?<owner>.*?)\/(?<name>.*?)[./]/gm;
+    const result = regex.exec(ws.status.gitStatus.cloneUrl);
+    if (!result) {
+        return ws.contextUrl.replace("https://", "");
+    }
+    const { host, owner, name } = result.groups!;
+    return `${host}/${owner}/${name}`;
```

Comment:
This looks a bit scary 😆. Could we instead fail or return an empty string if we cannot get it from the context URL? Not sure how this would support the deeply-nested GitLab repos, which are not just `/owner/repo`.

Reply:
I'm still not sure that we shoudl parse like that or add something like Repository on Workspace? cc @svenefftinge

Reply:
It looks good to me, and also is only used as a string to visualize.
I believe we have somewhere a function that translates the cloneURL to a shorter version.
IS this not just cutting off https:// and .git?

Reply:
Do you mean [`ContextURL.getNormalizedURL`](https://github.com/gitpod-io/gitpod/blob/907dc282ed493455f54137d787759539d46964d6/components/gitpod-protocol/src/context-url.ts#L30) ? I can't find other function that may have this feature. But it will only cut some built-in prefix out and won't trim `.git`.

I'll change to use `ContextURL.getNormalizedURL` here.

Reply:
@svenefftinge for comment above ☝️

Reply:
Ok

Reply:
We have a `RepoURL.parseRepoUrl` function in server that does approximation parsing of owner and name. If that's what's needed, we could thing of pulling this out to protocol. 

Otherwise, see `RepositoryURL` component in Dashboard if we just need a clean repo URL from a given clone URL.